scripts/build_climate_projections.py

```diff
@@ -0,0 +1,210 @@
+        dx_new,
+    )
+    newlat = np.arange(
+        round(min(cutout_xr.coords["y"].values), 1),
+        round(max(cutout_xr.coords["y"].values) + dx_new, 1),
+        dx_new,
+    )
+    cmip6_interp = cmip6_xr.interp(time=cmip6_xr.time, lat=newlat, lon=newlon)
+
+    return cmip6_interp
+
+
+# TODO fix years_window
+def subset_by_time(cmip6_xr, month, year, years_window):
+    # TODO add warning in case there are no enough years
+    cmip6_in_period = cmip6_xr.where(
+        (
+            (cmip6_xr["time.month"] == month)
+            & (cmip6_xr["time.year"] >= year - years_window)
+            & (cmip6_xr["time.year"] < year + years_window - 1)
+        ),
+        drop=True,
+    )
+    return cmip6_in_period
+
+
+def calculate_proj_of_average(cmip6_xr, month, year0, year1, years_window):
+    cmip6_interp_year0 = subset_by_time(
+        cmip6_xr,
+        month,
+        year=year0,
+    )
+    cmip6_interp_year1 = subset_by_time(cmip6_xr, month=month, year=year1)
+    dt_interp = cmip6_interp_year1["t"].mean("member").mean(
+        "time"
+    ) - cmip6_interp_year0["t"].mean("member").mean("time")
+    return dt_interp
+
+
+def build_projection_for_month(cutout_xr, dt_xr, month):
+    k_time = cutout_xr.time.dt.month.isin(month)
+
+    for i in range(0, len(cutout_xr.y)):
+        for j in range(0, len(cutout_xr.x)):
+            cutout_xr.temperature[k_time, i, j] = np.add(
+                cutout_xr.temperature[k_time, i, j],
+                np.array([dt_xr[i, j].item()] * k_time.sum().item()),
+            )
+
+    return cutout_xr
+
+
+def build_cutout_future(cutout_xr, cmip6_xr, months, year0, year1, years_window):
+    for k_month in [months]:
+        dt_k = calculate_proj_of_average(
+            cmip6_xr=cmip6_xr, month=k_month, year0=year0, year1=year1, years_window=5
+        )
+
+        build_projection_for_month(cutout_xr, dt_k, k_month)
+
+    cutout_xr = cutout_xr.where(cutout_xr.time.dt.month.isin(months), drop=True)
+
+    return cutout_xr
+
+
+if __name__ == "__main__":
+    if "snakemake" not in globals():
+        from _helpers import mock_snakemake
+
+        os.chdir(os.path.dirname(os.path.abspath(__file__)))
+        snakemake = mock_snakemake(
+            "build_climate_projections",
+            climate_scenario="ssp2-2.6",
+            present_year=2020,
+            future_year=2050,
+            years_window=5,
+            cutout="africa-2013-era5",
+        )
+    configure_logging(snakemake)
+
+    climate_scenario = snakemake.params.climate_scenario
+    present_year = snakemake.params.present_year
+    future_year = snakemake.params.future_year
+    years_window = snakemake.params.years_window
+    cutout = snakemake.input.cutout
+    cmip6 = snakemake.input.cmip6_avr
+
+    snapshots = pd.date_range(freq="h", **snakemake.params.snapshots)
+    season_in_focus = snapshots.month.unique().to_list()
+
+    cmip6_xr = xr.open_dataset(cmip6)
```

Comment:
How to load this dataset?

Reply:
As discussed, the dataset is available via Copernicus and licensed as CC-BY 4.0, being IPCC product 🙏🏽 

So, it can be downloaded manually or included into a dedicated databundle or loaded via Copernicus API with a request as simple as 
```python
c.retrieve(
    'projections-climate-atlas',
    {
        'format': 'zip',
        'origin': 'CMIP6',
        'experiment': 'ssp2_4_5',
        'domain': 'global',
        'period': '2015-2100',
        'variable': 'monthly_mean_of_daily_mean_temperature',
    },
    'download.zip')
```